src/Modules/OrchardCore.Commerce/Services/ProductService.cs

```diff
@@ -76,4 +76,18 @@ private static void FillField(ContentPart part, ContentPartFieldDefinition field
             part.Get(typeOfField, fieldName);
         }
     }
+
+    public string GetVariantKey(string sku)
+    {
+        var dashIndex = ProductServiceExtensions.GetDashIndex(sku);
+        return dashIndex == -1 ? sku : sku[(dashIndex + 1)..].ToUpperInvariant();
```

Comment:
Some time ago I've made a method for splitting a string into two parts, called `Partition` (name and concept shamelessly stolen from Python). I think if you don't have a dash here then something went wrong and it should throw.
So this better written as `sku.Partition("-").Right ?? throw new ArgumentException("The SKU doesn't contain a dash. Is it a product variant SKU?", nameof(sku))`.